rust/rbac-registration/src/cardano/cip509/mod.rs

```diff
@@ -92,7 +107,13 @@ pub(crate) enum Cip509IntIdentifier {
 impl Decode<'_, ()> for Cip509 {
     fn decode(d: &mut Decoder, ctx: &mut ()) -> Result<Self, decode::Error> {
```

Comment:
The metadata should take the paired transaction and witness set as context.  Its not possible to decode properly without it, i don't think?

Reply:
It is needed in validation and collecting the data for the registration chain.